scrapy/spidermiddlewares/referer.py

```diff
@@ -163,9 +163,10 @@ class StrictOriginPolicy(ReferrerPolicy):
     name = POLICY_STRICT_ORIGIN
 
     def referrer(self, response_url, request_url):
-        if ((self.tls_protected(response_url) and
-             self.potentially_trustworthy(request_url))
-            or not self.tls_protected(response_url)):
+        if (
+            self.tls_protected(response_url) and self.potentially_trustworthy(request_url)
+            or not self.tls_protected(response_url)
```

Comment:
Is it the same condition? Tbh I found a previous one, with (), easier to understand.

Reply:
It is, because of the [operator precedence](https://docs.python.org/3/reference/expressions.html#operator-precedence). It seems like the `black` developers feel as you do:

```python
def referrer(self, response_url, request_url):
    if (
        self.tls_protected(response_url) and self.potentially_trustworthy(request_url)
    ) or not self.tls_protected(response_url):
        return self.origin_referrer(response_url)
```

I guess it's a matter of personal preference. I'd argue that the parenthesis are not strictly necessary, the same way they are redundant in `(2 * 3) + 4^5`. In that sense, I'd prefer to avoid them, but the truth is that I don't think they hurt legibility either.

Reply:
I’m OK with the changes given the and-joined statements remain in the same line, but given the line length I guess it may indeed be more readable to keep the and-joined statements in separate lines, within parenthesis and an additional level of indentation.

Reply:
Is line length an issue? The proposed change goes up to 90, and we have an approved PR to increase it to 119.